keps/sig-network/4427-relaxed-dns-search-validation/README.md

```diff
@@ -0,0 +1,473 @@
+-->
+
+##### e2e tests
+
+- Add a test that verifies successful creation of a pod whose `dnsConfig.searches` contains an underscore
+- Add tests that verify successful creation of objects with a podTemplate whose `dnsConfig.searches`
+  contains an underscore
+
+### Graduation Criteria
+
+#### Alpha
+- Feature implemented behind a gate
+- Initial e2e tests completed and enabled
+
+#### Beta
+- No trouble reports from alpha release
+
+#### GA
+- No trouble reports with the beta release, plus some anecdotal evidence of it being used successfully.
+
+### Upgrade / Downgrade Strategy
+
+See Proposal section.
+
+### Version Skew Strategy
+
+Kubelet only checks size limits but otherwise passes values through
+[source](https://github.com/kubernetes/kubernetes/blob/f025a96d2f60984765731e01ad0de2c89e959b42/pkg/kubelet/network/dns/dns.go#L114).
+
+Since the resolv.conf file is interpreted by the DNS resolver in the container image and not by the container runtime, the change
+does not depend on the container runtime or its version.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: RelaxedDNSSearchValidation
+  - Components depending on the feature gate: kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+No, there is no change to default behavior.
+This is a change to validation.
```

Comment:
nit: validation is behavior, so this is a yes, but we're ratcheting in a widening of validation, which is OK